pkg/ottl/ottlfuncs/func_trim.go

```diff
@@ -0,0 +1,42 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type TrimArguments[K any] struct {
+	Target      ottl.StringGetter[K]
+	Replacement ottl.Optional[string]
+}
+
+func NewTrimFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("Trim", &TrimArguments[K]{}, createTrimFunction[K])
+}
+
+func createTrimFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*TrimArguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("TrimFactory args must be of type *TrimArguments[K]")
+	}
+
+	return trim(args.Target, args.Replacement), nil
+}
+
+func trim[K any](target ottl.StringGetter[K], replacement ottl.Optional[string]) ottl.ExprFunc[K] {
+        replacementString := replacement.Get()
```

Comment:
@rnishtala-sumo we do need to still check if `replacement` is empty or not.  If `IsEmpty()` returns true then we need to use a default cut string of `" "`. If `IsEmpty()` returns false then we can use whatever the value of `Get` returns.